src/components/ProfileMenu.js

```diff
@@ -21,6 +21,41 @@ export default function ProfileMenu({
     handleLogout,
   } = useContext(UserContext);
   const navigate = useNavigate();
+  const modalRef = useRef();
+
+  useEffect(() => {
+    function keyListener(e) {
+      if (e.keyCode === 27) {
+        setIsProfileMenuOpen(false);
+      } else if (e.keyCode === 9 || e.keyCode === "Tab") {
+        handleTabKey(e);
+      }
+    }
+    document.addEventListener("keydown", keyListener);
+    return () => document.removeEventListener("keydown", keyListener);
+  });
+
+  const handleTabKey = (e) => {
+    const focusableModalElements = modalRef.current?.querySelectorAll("button");
+    const firstElement = focusableModalElements[0];
+    const lastElement =
+      focusableModalElements[focusableModalElements.length - 1];
+    let isButtonfocused = false;
+    for (let i = 0; i <= focusableModalElements.length; i++) {
+      if (focusableModalElements[i] === document.activeElement) {
+        isButtonfocused = true;
+      }
+    }
+    !isButtonfocused && firstElement.focus();
+    if (!e.shiftKey && document.activeElement === lastElement) {
+      firstElement.focus();
+      return e.preventDefault();
+    }
+    if (e.shiftKey && document.activeElement === firstElement) {
+      lastElement.focus();
+      e.preventDefault();
+    }
+  };
```

Comment:
As long as you're not calling the function anywhere else but in useEffect, you can move it up to live inside the useEffect callback.